src/components/KEmptyState/KEmptyState.vue

```diff
@@ -1,143 +1,173 @@
-      >
-        <slot name="message" />
+      <div class="empty-state-icon">
+        <slot name="icon">
+          <component
+            :is="getEmptyStateIcon"
+            class="alert-icon"
+            :color="getIconColor"
+            :size="KUI_ICON_SIZE_60"
+          />
+        </slot>
       </div>
+      <span
+        v-if="title"
+        class="empty-state-title"
+      >
+        {{ title }}
+      </span>
       <div
-        v-if="$slots.cta || (!ctaIsHidden && ctaText)"
-        class="k-empty-state-cta"
+        v-if="message || $slots.default"
+        class="empty-state-message"
       >
-        <slot name="cta">
-          <KButton
-            appearance="primary"
-            size="small"
-            @click.prevent="() => handleClick && handleClick()"
-          >
-            {{ ctaText }}
-          </KButton>
+        <slot name="default">
+          <p>
+            {{ message }}
+          </p>
         </slot>
       </div>
     </div>
-  </section>
+    <div
+      v-if="actionButtonVisible && (actionButtonText || $slots.action)"
+      class="empty-state-action"
+    >
+      <slot name="action">
+        <KButton
+          :disabled="actionButtonDisabled"
+          @click="$emit('action-click')"
+        >
+          {{ actionButtonText }}
+        </KButton>
+      </slot>
+    </div>
+  </div>
 </template>
 
 <script lang="ts" setup>
+import { computed, type PropType } from 'vue'
+import { AnalyticsIcon, WarningIcon, CogIcon, FileEmptyIcon, SearchIcon, KongIcon } from '@kong/icons'
+import { KUI_COLOR_TEXT_NEUTRAL, KUI_COLOR_TEXT_WARNING, KUI_ICON_SIZE_60 } from '@kong/design-tokens'
 import KButton from '@/components/KButton/KButton.vue'
-import KIcon from '@/components/KIcon/KIcon.vue'
-import { KUI_COLOR_TEXT_NEUTRAL_STRONGEST, KUI_ICON_SIZE_80 } from '@kong/design-tokens'
+import { EmptyStateIconVariants } from '@/types'
+import type { EmptyStateIconVariant } from '@/types'
 
-defineProps({
-  isError: {
-    type: Boolean,
-    default: false,
-  },
-  iconSize: {
+type EmptyStateIcon = typeof AnalyticsIcon // all icons are the same type so we can use any of them
+
+const props = defineProps({
+  title: {
     type: String,
-    default: KUI_ICON_SIZE_80,
+    default: '',
   },
-  icon: {
+  message: {
     type: String,
     default: '',
   },
-  ctaIsHidden: {
+  actionButtonVisible: {
     type: Boolean,
-    default: false,
+    default: true,
   },
-  ctaText: {
+  actionButtonText: {
     type: String,
     default: '',
   },
-  handleClick: {
-    type: Function,
-    default: null,
-  },
-  iconColor: {
-    type: String,
-    default: '',
+  actionButtonDisabled: {
+    type: Boolean,
+    default: false,
   },
-  iconSecondaryColor: {
-    type: String,
-    default: '',
+  iconVariant: {
+    type: String as PropType<EmptyStateIconVariant>,
+    default: EmptyStateIconVariants.Default,
   },
 })
-</script>
 
-<style lang="scss" scoped>
+defineEmits(['action-click'])
```

Comment:
can this be `click-action` instead to be closer to a native click, e.g. `@click-action`